packages/components/molecules/f-skeleton-loader/src/components/SkeletonLoader.vue

```diff
@@ -0,0 +1,75 @@
+
+
+<script>
+export default {
+    name: 'SkeletonLoader',
+    components: {},
+    props: {
+        skeletonType: {
+            type: String,
+            default: 'heading'
+        }
+    },
+    data () {
+        return {
+            rootTypes: {
+                text: 'text',
+                heading: 'heading'
+            }
+        };
+    },
+    methods: {
+        genBone (text = this.skeletonType, children) {
+            return this.$createElement(
+                'div',
+                {
+                    staticClass: `${this.$style[`c-skeleton-loader__${text}`]} ${this.$style['c-skeleton-loader__bone']}`
+                },
+                children
+            );
+        }
+    },
+    render (createElement) {
+        return createElement(
```

Comment:
Any reason why you're not using the `template` to generate these elements?

Reply:
Yes the next PRs will extend the component to dynamically render groups of nested divs. Therefore its using createElement to do this programmatically.

Reply:
I might be missing something here, but I still don't quite understand why just using `template` from the beginning is not an option, sorry! Do you mind trying to explain a bit more please, @AndyMayJE?

Reply:
The component is never going to render anything other than dynamically generated nested divs based on the skeletonType prop. Expressing this in template logic could be tricky as nesting may be multiple levels deep. The template would be hard to understand and not really provide a visual representation of what may be rendered. 

A simple article will look like:

```
<div class="c-skeleton-loader-article c-skeleton-loader-bone">
    <div class="c-skeleton-loader-heading c-skeleton-loader-bone"></div>
    <div class="c-skeleton-loader-paragraph c-skeleton-loader-bone">
        <div class="c-skeleton-loader-text c-skeleton-loader-bone"></div>
        <div class="c-skeleton-loader-text c-skeleton-loader-bone"></div>
        <div class="c-skeleton-loader-text c-skeleton-loader-bone"></div>
    </div>
</div>
```

Hope that makes sense? Happy to change approach if you have alternative ideas?

Reply:
What I had in mind was that we'd have various `.vue` components within `f-skeleton-loader` and that we'd load the corresponding one based on the `prop`. 

For instance, `SkeletonLoader.vue` gets passed in `type="article"` and then it would render an `Article.vue` component what would have the template you described above.

But feel free to carry on with your approach and then we can review it as the component grows in complexity and requirements 😄

Reply:
@SachaZvetelman ok that makes sense thanks. Certainly not 100% set on this approach it might be over complicating what we need. I am going to spend an hour or so investigating. I guess the disadvantage might be managing tiny child components. I did consider and look at using JSX in Vue to express the skeletons.

Reply:
@SachaZvetelman @DamianMullins switched back to <template> in favour of a more simple approach which is all we need for now. Next PRs will introduce small skeleton components that will be loaded based on the skeleton prop.